src/topp/GNPSExport.cpp

```diff
@@ -79,297 +147,368 @@ using namespace std;
 class TOPPGNPSExport : public TOPPBase
 {
 public:
-  TOPPGNPSExport() :
-  TOPPBase("GNPSExport", "Tool to export consensus features into MGF format", false) {}
+  TOPPGNPSExport() : TOPPBase(
+    "GNPSExport", 
+    "Tool to export representative consensus MS/MS scan per consensusElement into a .MGF file format.\nSee the documentation on https://ccms-ucsd.github.io/GNPSDocumentation/featurebasedmolecularnetworking_with_openms", 
+    true,
+    {
+      { 
+        "Nothias L.F. et al.", // authors
+        "Feature-based Molecular Networking in the GNPS Analysis Environment", // title
+        "bioRxiv 812404 (2019)", // when_where
+        "10.1101/812404" // doi
+      }
+    }
+  ) {}
+
 
 private:
-  double DEF_COSINE_SIMILARITY = 0.95;
-  double DEF_PRECURSOR_MZ_TOLERANCE = 0.0001;
-  double DEF_PRECURSOR_RT_TOLERANCE = 5;
+  static constexpr double DEF_COSINE_SIMILARITY = 0.9;
+  static constexpr double DEF_MERGE_BIN_SIZE = static_cast<double>(BinnedSpectrum::DEFAULT_BIN_WIDTH_HIRES);
+
+  static constexpr double DEF_PREC_MASS_TOL = 0.5;
+  static constexpr bool DEF_PREC_MASS_TOL_ISPPM = false;
+
+  static constexpr double DEF_PEPT_CUTOFF = 5;
+  static constexpr double DEF_MSMAP_CACHE = 50;
+
+
+  void writeMS2BlockToFile(
+    ofstream& output_file,
+    const map<double,int>& ms2_block, 
+    const String& output_type,
+    const int& scan_index,
+    const String& feature_id,
+    const int& feature_charge,
+    const String& feature_mz,
+    const String& spec_index,
+    const String& feature_rt
+  ) 
+  {
+    output_file << "BEGIN IONS" << "\n";
+    output_file << "OUTPUT=" << output_type << "\n";
```

Comment:
hint/optional:
for better readability you can do
```
output_file << "BEGIN IONS" << "\n"
            << "OUTPUT=" << output_type << "\n";
```
i.e. omit repeated usage of the stream and indent accordingly